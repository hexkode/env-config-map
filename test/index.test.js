const envConfigMap = require('../index.js');

test('baseline test equal expected output', () => {
  process.env.NODE_ENV = 'test';
  process.env.SERVER_PORT = '8080';
  process.env.ENABLE_CORS = 'true';
  process.env.DB_PASSWORD = 'mypassword';
  process.env.DB_ENABLE_PROFILER = 'yes';
  process.env.EXAMPLE_OBJECT = '{ "retry": 3, "timeout": 1000 } ';
  process.env.EXAMPLE_ARRAY = '[ "a", 1 ]';
  process.env.EXAMPLE_ARRAY_COMMA_DELIM = 'id,email,   dateCreated   ,dateModified';
  process.env.NOT_IN_CONFIG_MAP = 'not mapped';

  const configMap = {
    NODE_ENV: { default: 'development' },
    LOG_LEVEL: { default: 'info' },
    SERVER_ADDRESS: {},
    SERVER_PORT: { default: 80, type: 'number' },
    ENABLE_CORS: { default: false, type: 'boolean' },
    DB_PASSWORD: { isSecret: true },
    DB_ENABLE_PROFILER: { default: false, type: 'yesNoBool' },
    EXAMPLE_OBJECT: { type: 'object' },
    EXAMPLE_ARRAY: { type: 'object' },
    EXAMPLE_ARRAY_COMMA_DELIM: { type: 'arrayCommaDelim' },
  };

  const options = {
    redactedString: 'XXXXXXXXXX',
    typeTransform: {
      yesNoBool: string => (string === 'yes' ? true : false),
    },
  };

  const config = envConfigMap(configMap, options);

  expect(config).toMatchSnapshot();
  expect(config.getRedacted()).toMatchSnapshot();
  expect(config.getDefaultOptions()).toMatchSnapshot();
  expect(config.getOptions()).toMatchSnapshot();
});
