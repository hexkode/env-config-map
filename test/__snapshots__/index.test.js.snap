// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`baseline test equal expected output 1`] = `
Object {
  "DB_ENABLE_PROFILER": true,
  "DB_PASSWORD": "mypassword",
  "ENABLE_CORS": true,
  "EXAMPLE_ARRAY": Array [
    "a",
    1,
  ],
  "EXAMPLE_ARRAY_COMMA_DELIM": Array [
    "id",
    "email",
    "dateCreated",
    "dateModified",
  ],
  "EXAMPLE_OBJECT": Object {
    "retry": 3,
    "timeout": 1000,
  },
  "LOG_LEVEL": "info",
  "MISSING_TYPE_TRANSFORM": "pass thru          ",
  "NODE_ENV": "test",
  "SERVER_ADDRESS": undefined,
  "SERVER_PORT": 8080,
  "getDefaultOptions": [Function],
  "getOptions": [Function],
  "getRedacted": [Function],
}
`;

exports[`baseline test equal expected output 2`] = `
Object {
  "DB_ENABLE_PROFILER": true,
  "DB_PASSWORD": "XXXXXXXXXX",
  "ENABLE_CORS": true,
  "EXAMPLE_ARRAY": Array [
    "a",
    1,
  ],
  "EXAMPLE_ARRAY_COMMA_DELIM": Array [
    "id",
    "email",
    "dateCreated",
    "dateModified",
  ],
  "EXAMPLE_OBJECT": Object {
    "retry": 3,
    "timeout": 1000,
  },
  "LOG_LEVEL": "info",
  "MISSING_TYPE_TRANSFORM": "pass thru          ",
  "NODE_ENV": "test",
  "SERVER_ADDRESS": undefined,
  "SERVER_PORT": 8080,
}
`;

exports[`baseline test equal expected output 3`] = `
Object {
  "getEnv": [Function],
  "redactedString": "**********",
  "typeTransform": Object {
    "_normalize": [Function],
    "arrayCommaDelim": [Function],
    "boolean": [Function],
    "number": [Function],
    "object": [Function],
    "string": [Function],
  },
}
`;

exports[`baseline test equal expected output 4`] = `
Object {
  "getEnv": [Function],
  "redactedString": "XXXXXXXXXX",
  "typeTransform": Object {
    "_normalize": [Function],
    "arrayCommaDelim": [Function],
    "boolean": [Function],
    "number": [Function],
    "object": [Function],
    "string": [Function],
    "yesNoBool": [Function],
  },
}
`;
